Source/Scene/modelMaterialsCommon.js

```diff
@@ -0,0 +1,792 @@
+                }
+                else {
+                    fragmentShader += '  vec4 diffuse = u_diffuse;\n';
+                }
+                fragmentShader += '  vec3 diffuseLight = vec3(0.0, 0.0, 0.0);\n';
+                colorCreationBlock += '  color += diffuse.rgb * diffuseLight;\n';
+            }
+
+            if (hasSpecular) {
+                if (techniqueParameters.specular.type === WebGLConstants.SAMPLER_2D) {
+                    fragmentShader += '  vec3 specular = texture2D(u_specular, ' + v_texcoord + ');\n';
+                }
+                else {
+                    fragmentShader += '  vec3 specular = u_specular.rgb;\n';
+                }
+                fragmentShader += '  vec3 specularLight = vec3(0.0, 0.0, 0.0);\n';
+                colorCreationBlock += '  color += specular * specularLight;\n';
+            }
+
+            if (defined(techniqueParameters.transparency)) {
+                finalColorComputation = '  gl_FragColor = vec4(color * diffuse.a, diffuse.a * u_transparency);\n';
+            }
+            else {
+                finalColorComputation = '  gl_FragColor = vec4(color * diffuse.a, diffuse.a);\n';
+            }
+        }
+        else {
+            if (defined(techniqueParameters.transparency)) {
+                finalColorComputation = '  gl_FragColor = vec4(color, u_transparency);\n';
+            }
+            else {
+                finalColorComputation = '  gl_FragColor = vec4(color, 1.0);\n';
+            }
+        }
+
+        if (defined(techniqueParameters.emission)) {
+            if (techniqueParameters.emission.type === WebGLConstants.SAMPLER_2D) {
+                fragmentShader += '  vec3 emission = texture2D(u_emission, ' + v_texcoord + ');\n';
+            }
+            else {
+                fragmentShader += '  vec3 emission = u_emission.rgb;\n';
+            }
+            colorCreationBlock += '  color += emission;\n';
+        }
+
+        if (defined(techniqueParameters.ambient)) {
+            if (techniqueParameters.ambient.type === WebGLConstants.SAMPLER_2D) {
+                fragmentShader += '  vec3 ambient = texture2D(u_ambient, ' + v_texcoord + ');\n';
+            }
+            else {
+                fragmentShader += '  vec3 ambient = u_ambient.rgb;\n';
+            }
+        }
+        else {
+            fragmentShader += '  vec3 ambient = diffuse.rgb;\n';
+        }
+        fragmentShader += '  vec3 viewDir = -normalize(v_positionEC);\n';
+        fragmentShader += '  vec3 ambientLight = vec3(0.0, 0.0, 0.0);\n';
+        colorCreationBlock += '  color += ambient * ambientLight;\n';
+
+        // Add in light computations
+        fragmentShader += fragmentLightingBlock;
+
+        fragmentShader += colorCreationBlock;
+        fragmentShader += finalColorComputation;
+        fragmentShader += '}\n';
+
+        // TODO: Handle texture transparency
```

Comment:
Yep. See https://github.com/KhronosGroup/glTF/issues/424#issuecomment-148774205 for my proposed solution.